### PR TITLE
- add dc1394

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **dc1394/2.2.6**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libdc1394/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            if self.options.shared:
+                args = ["--disable-static", "--enable-shared"]
+            else:
+                args = ["--disable-shared", "--enable-static"]
+            self._env_build.configure(args=args)
+        return self._env_build
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows is not supported")
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:37]: It looks like only Visual Studio needs special treatment:
https://sourceforge.net/p/libdc1394/code/ci/master/tree/INSTALL-windows

File: recipes/libdc1394/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            os.chdir("bin")
+            self.run(".%sexample" % os.sep)
```
Comment (madebr) [inline recipes/libdc1394/all/test_package/conanfile.py:18]:
```suggestion
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

File: recipes/libdc1394/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            env_build.install()
+
+        self.copy(pattern="COPYING", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+
+    def package_info(self):
+        self.cpp_info.libs = ["dc1394"]
+        if self.settings.os == "Macos":
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:68]:
```suggestion
        if tools.os_info.is_apple_os(self.settings.os):
```

File: recipes/libdc1394/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            env_build = self._configure_autotools()
+            env_build.install()
+
+        self.copy(pattern="COPYING", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+
+    def package_info(self):
+        self.cpp_info.libs = ["dc1394"]
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:67]: https://sourceforge.net/p/libdc1394/code/ci/master/tree/libdc1394-2.pc.in
```suggestion
    def package_info(self):
        self.cpp_info.names["pkg_config"] = "libdc1394-2.pc"
        self.cpp_info.libs = ["dc1394"]
```

File: recipes/libdc1394/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            self._env_build = AutoToolsBuildEnvironment(self)
+            if self.options.shared:
+                args = ["--disable-static", "--enable-shared"]
+            else:
+                args = ["--disable-shared", "--enable-static"]
+            self._env_build.configure(args=args)
+        return self._env_build
+
+    def configure(self):
+        if self.settings.os == "Windows":
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:36]:
```suggestion
    def configure(self):
        if self.options.shared:
            del self.options.fPIC
        if self.settings.os == "Windows":
```

File: recipes/libdc1394/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class Libdc1394Conan(ConanFile):
+    name = "libdc1394"
+    version = "2.2.6"
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:9]:
```suggestion
```

File: recipes/libdc1394/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    license = "LGPL-2.1"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = 'https://damien.douxchamps.net/ieee1394/libdc1394/'
+    description = "libdc1394 provides a complete high level API to control IEEE 1394 based cameras"
+    topics = ("conan", "ieee1394", "camera", "iidc", "dcam")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:18]:
```suggestion
```

File: recipes/libdc1394/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    _env_build = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_autotools(self):
+        if not self._env_build:
+            self._env_build = AutoToolsBuildEnvironment(self)
+            if self.options.shared:
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:28]: The configure script looks for libusb, sdl and opengl.
Perhaps optionally add the libusb requirement and disable SDL2 to have a reproducible build?
The configure script has also a ´--disable-examples´ option.

File: recipes/libdc1394/all/test_package/test_package.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+{
+	dc1394_t *dc1394 = NULL;
+	dc1394error_t error;
+	dc1394camera_list_t *cameras = NULL;
+
+	dc1394 = dc1394_new();
+	if (!dc1394)
+	{
+		fprintf(stderr, "cannot create dc1394!\n");
+		return EXIT_FAILURE;
```
Comment (madebr) [inline recipes/libdc1394/all/test_package/test_package.c:16]:
```suggestion
		fprintf(stderr, "cannot create dc1394!\n");
		fprintf(stderr, "This is a test package, so ignore!\n");
		return EXIT_SUCCESS
```

File: recipes/libdc1394/all/test_package/test_package.c [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+	}
+	error = dc1394_camera_enumerate(dc1394, &cameras);
+	if ((error > 0) || (error <= -DC1394_ERROR_NUM))
+	{
+		fprintf(stderr, "failed to enumerate cameras!\n");
+		dc1394_free(dc1394);
+		return EXIT_FAILURE;
```
Comment (madebr) [inline recipes/libdc1394/all/test_package/test_package.c:23]:
```suggestion
		fprintf(stderr, "This is a test package, so ignore! (better luck next time)");
		return EXIT_SUCCESS;
```

File: recipes/libdc1394/all/test_package/test_package.c [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+		fprintf(stderr, "cannot create dc1394!\n");
+		fprintf(stderr, "This is a test package, so ignore!\n");
+		return EXIT_SUCCESS
+	}
+	error = dc1394_camera_enumerate(dc1394, &cameras);
+	if ((error > 0) || (error <= -DC1394_ERROR_NUM))
+	{
+		fprintf(stderr, "failed to enumerate cameras!\n");
+		dc1394_free(dc1394);
+		fprintf(stderr, "This is a test package, so ignore! (better luck next time)");
```
Comment (madebr) [inline recipes/libdc1394/all/test_package/test_package.c:24]:
```suggestion
		fprintf(stderr, "This is a test package, so ignore! (better luck next time)\n");
```

File: recipes/libdc1394/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:3]:
```suggestion
```

File: recipes/libdc1394/all/conanfile.py [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+            env_build = self._configure_autotools()
+            env_build.install()
+
+        self.copy(pattern="COPYING", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "libdc1394-2.pc"
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:67]:
```suggestion
        self.cpp_info.names["pkg_config"] = "libdc1394-{}.pc".format(tools.Version(self.version).major)
```

File: recipes/libdc1394/all/conanfile.py [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+        self.cpp_info.libs = ["dc1394"]
+        if tools.os_info.is_apple_os(self.settings.os):
```
Comment (madebr) [inline recipes/libdc1394/all/conanfile.py:69]:
```suggestion
        if tools.is_apple_os(self.settings.os):
```

File: recipes/libdc1394/all/conanfile.py [hunk 15]
```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+import os
+
+
+class Libdc1394Conan(ConanFile):
+    name = "libdc1394"
+    license = "LGPL-2.1"
```
Comment (uilianries) [inline recipes/libdc1394/all/conanfile.py:9]:
```suggestion
    license = "LGPL-2.1-or-later"
```

LGPL-2.1 is deprecated. dc1394 license is "-or-later".

https://spdx.org/licenses/LGPL-2.1-or-later.html

File: recipes/libdc1394/all/test_package/test_package.c [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+	dc1394_t *dc1394 = NULL;
+	dc1394error_t error;
+	dc1394camera_list_t *cameras = NULL;
+
+	dc1394 = dc1394_new();
+	if (!dc1394)
+	{
+		fprintf(stderr, "cannot create dc1394!\n");
+		fprintf(stderr, "This is a test package, so ignore!\n");
+		return EXIT_SUCCESS
```
Comment (SSE4) [inline recipes/libdc1394/all/test_package/test_package.c:17]:
```suggestion
		return EXIT_SUCCESS;
```

File: recipes/libdc1394/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/libdc1394/all/conanfile.py:36]: it requires a driver, so I give up for now with VS :)

File: recipes/libdc1394/all/conanfile.py [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_autotools(self):
+        if not self._env_build:
+            self._env_build = AutoToolsBuildEnvironment(self)
+            if self.options.shared:
+                args = ["--disable-static", "--enable-shared"]
+            else:
+                args = ["--disable-shared", "--enable-static"]
```
Comment (SSE4) [inline recipes/libdc1394/all/conanfile.py:28]:
```suggestion
                args = ["--disable-shared", "--enable-static"]
            args.extend(["--disable-examples"])
```

**Review by uilianries — DISMISSED** [review]
Tested on Linux. LGTM

**Review by prince-chrismc — DISMISSED** [review]

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libdc1394/all/conanfile.py (hunk 8 above)
Comment (SSE4) [reply recipes/libdc1394/all/conanfile.py:25]: added `--disable-examples`

**Review by danimtb — APPROVED** [review]

**Pull request merged**